src/world.rs

```diff
@@ -0,0 +1,214 @@
+                        path: path.to_owned(),
+                        err: "Failed to get file name".into(),
+                    })?
+                    .to_str()
+                    .ok_or_else(|| Error::ResourceLoadingError {
+                        path: path.to_owned(),
+                        err: "Failed to convert file name to string".into(),
+                    })?
+                    .to_owned();
+
+                let captures = re.captures(&filename).unwrap();
+
+                let x = captures
+                    .get(1)
+                    .ok_or_else(|| Error::ResourceLoadingError {
+                        path: path.to_owned(),
+                        err: format!("Failed to parse x pattern from file {}", filename).into(), 
+                    })?
+                    .as_str()
+                    .parse::<i32>()
+                    .map_err(|e| Error::ResourceLoadingError {
+                        path: path.to_owned(),
+                        err: Box::new(e),
+                    })?;
+
+                let x = match x
+                    .checked_mul(pattern.multiplier_x as i32)
+                    .and_then(|x| x.checked_add(pattern.offset_x))
+                {
+                    Some(x) => x,
+                    None => {
+                        return Err(Error::ResourceLoadingError {
+                            path: path.to_owned(),
+                            err: "Arithmetic Overflow on multiplierX and offsetX".into(),
+                        })
+                    }
+                };
+                let y = captures
+                    .get(2)
+                    .ok_or_else(|| Error::ResourceLoadingError {
+                        path: path.to_owned(),
+                        err: format!("Failed to parse y pattern from file {}", filename).into(),
+                    })?
+                    .as_str()
+                    .parse::<i32>()
+                    .map_err(|e| Error::ResourceLoadingError {
+                        path: path.to_owned(),
+                        err: Box::new(e),
+                    })?;
+                let y = match y
+                    .checked_mul(pattern.multiplier_y as i32)
+                    .and_then(|y| y.checked_add(pattern.offset_y))
+                {
+                    Some(y) => y,
+                    None => {
+                        return Err(Error::ResourceLoadingError {
+                            path: path.to_owned(),
+                            err: "Arithmetic Overflow on multiplierY and offsetY".into(),
+                        })
+                    }
+                };
+                Ok(WorldMap {
+                    filename,
+                    x,
+                    y,
+                    width: Some(pattern.multiplier_x),
+                    height: Some(pattern.multiplier_y),
+                })
+            })
+            .collect::<Vec<_>>();
```

Comment:
I think you can collect into a `Result<Vec<_>, _>` here instead, and use the try operator right away. That way you can also exchange the for loop after this with just a `maps.append(files)`.